photoframe/source/app/src/main/java/org/kaaproject/kaa/demo/photoframe/fragment/BaseFragment.java

```diff
@@ -0,0 +1,248 @@
+            return;
+        }
+
+        FragmentManager fm = mActivity.getSupportFragmentManager();
+        FragmentTransaction ft = fm.beginTransaction();
+        ft.replace(R.id.container, fragment, fragment.getFragmentTag());
+        ft.setTransition(FragmentTransaction.TRANSIT_FRAGMENT_OPEN);
+        ft.addToBackStack(fragment.getFragmentTag());
+        ft.commit();
+    }
+
+    public void move(Activity mActivity) {
+        move((MainActivity) mActivity, this);
+    }
+
+    public static void popBackStack(FragmentActivity activity) {
+        if (activity == null) {
+            Log.e(PhotoFrameConstants.LOG_TAG, "Unable pop fragment. Invalid args.");
+            return;
+        }
+
+        FragmentManager fm = activity.getSupportFragmentManager();
+        if (fm.getBackStackEntryCount() > 0) {
+            fm.popBackStack();
+        }
+    }
+
+
+    @Override
+    public void onAttach(Activity activity) {
+        super.onAttach(activity);
+
+        if (activity != null) {
+            mActionBar = ((MainActivity) activity).getSupportActionBar();
+            manager = ((MainActivity) activity).getManager();
+        }
+
+        manager.registerEventBus(this);
+    }
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+
+        if (context == null) {
+            mActionBar = ((MainActivity) context).getSupportActionBar();
+            manager = ((MainActivity) context).getManager();
+        }
+
+        manager.registerEventBus(this);
+    }
+
+
+    @Override
+    public void onResume() {
+        super.onResume();
+
+        if (updateActionBar() && mActionBar != null) {
+            int options = ActionBar.DISPLAY_SHOW_TITLE;
+            if (displayHomeAsUp())
+                options |= ActionBar.DISPLAY_HOME_AS_UP;
+
+            mActionBar.setDisplayOptions(options, ActionBar.DISPLAY_HOME_AS_UP
+                    | ActionBar.DISPLAY_SHOW_TITLE);
+            mActionBar.setTitle(getTitle());
+
+            mActionBar.setDisplayShowTitleEnabled(true);
+            mActionBar.setHomeButtonEnabled(displayHomeAsUp());
+        }
+
+        manager.registerEventBus(this);
+    }
+
+
+    @Override
+    public void onPause() {
+        super.onPause();
+
+        manager.unregisterEventBus(this);
+    }
+
+
+    @Subscribe
+    public void onEvent(final Events.PlayAlbumEvent playAlbumEvent) {
+        final Fragment fragment = BaseFragment.getCurrentFragment((MainActivity) getActivity());
+
+        getActivity().runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+
+                if (manager.isUserAttached()) {
+                    loadSlideshow(playAlbumEvent, fragment);
+                } else {
+                    // If you logout, but get this event
+                    Toast.makeText(getActivity(), R.string.logout_interaction_event, Toast.LENGTH_SHORT).show();
```

Comment:
Please remove the whole if-else clause. Leave only loadSlideshow invocation. Please read  [Endpoints messaging](http://docs.kaaproject.org/display/KAA/Messaging+across+endpoints) (take attention on [Attach endpoint to user section](http://docs.kaaproject.org/display/KAA/Messaging+across+endpoints#Messagingacrossendpoints-Attachendpointtouser)) and [Attaching endpoints to users](http://docs.kaaproject.org/display/KAA/Attaching+endpoints+to+users) in order to understand how this feature works.